docs/api/spec/query-api.yaml

```diff
@@ -0,0 +1,288 @@
+# The openapi specification for the like API
+openapi: 3.0.0
+info:
+  title: Treetracker Like API
+  contact: {}
+  version: '1.0.0'
+servers:
+  - url: https://localhost:3010
+    variables:
+      environment:
+        default: dev
+        enum:
+          - dev     # Development server
+          - test    # Test server
+security:
+  - OAuth2:
+      - read
+      - write
+      - admin
```

Comment:
We use OAuth2 (Keycloak is an implementation of oauth2) to do the auth stuff for our API, and on Keycloak, we will define the access model like this:

* Resources:
  * /treetracker-like/types/
  * /treetracker-like/types/{type_id}
  * /treetracker-like/types/{types_uuid}/things/{object_uuid}
  * /treetracker-like/types/
* Scopes 
  * read
  * write
  * admin

So to be able to like a tree, the user should get `read` permission against `/treetracker-like/types/{types_uuid}/things/{object_uuid}` here `object_uuid` is the tree id.
On the admin panel (the like dashboard), we can manage `types`, for an admin user, they should have `admin` permission aganst `/treetracker-like/types/` resource, so they can do `creating a type` operation on the admin panel.